### PR TITLE
Add simple smoke test for OSS Dockerfile

### PR DESCRIPTION
**Pull request by ldziedziul**
This PR adds workflow testing OSS Dockerfile using our CLC application.

Example run: https://github.com/ldziedziul/hazelcast-docker/actions/runs/3565826014/jobs/5991476430

### CODE REVIEW

File: .github/scripts/simple-smoke-test.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+set -e
+set -o pipefail
+
+export CLV_VERSION=v5.2.0-beta3-preview1
```
Comment (frant-hartm) [inline .github/scripts/simple-smoke-test.sh:6]: CLV_VERSION?

File: .github/scripts/simple-smoke-test.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+function test_docker_image() {
+    local image=$1
+    local container_name=$2
+    echo "Starting container '$container_name' from image '$image'"
+    docker run -it --name "$container_name" -d -p5701:5701 "$image"
```
Comment (frant-hartm) [inline .github/scripts/simple-smoke-test.sh:12]: I am not sure where this is stored but maybe add `--rm` so the container is removed after being stopped.

**Review by frant-hartm — CHANGES_REQUESTED** [review]

File: .github/scripts/simple-smoke-test.sh (hunk 2 above)
Comment (ldziedziul) [reply .github/scripts/simple-smoke-test.sh:12]: I've skipped '--rm' to be able to get logs and store them as artifacts. GitHub runners are cleared before each reuse

File: .github/scripts/simple-smoke-test.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    docker stop "$container_name"
+
+    if [ "$expected" != "$actual" ]; then
+        echo "Expected to read '${expected}' but got '${actual}'"
+        exit 1;
+    fi
+}
+
+function install_clc() {
+  CLC_URL="https://github.com/hazelcast/hazelcast-commandline-client/releases/download/${CLV_VERSION}/hazelcast-clc_${CLV_VERSION}_linux_amd64.tar.gz"
```
Comment (frant-hartm) [inline .github/scripts/simple-smoke-test.sh:30]: Here as well.
```suggestion
  CLC_URL="https://github.com/hazelcast/hazelcast-commandline-client/releases/download/${CLC_VERSION}/hazelcast-clc_${CLV_VERSION}_linux_amd64.tar.gz"
```

**Review by frant-hartm — APPROVED** [review]

**Pull request merged**